hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapReplicationStateHolder.java

```diff
@@ -99,10 +116,52 @@ void prepare(PartitionContainer container, Collection<ServiceNamespace> namespac
                 recordSet.add(recordReplicationInfo);
             }
             data.put(mapName, recordSet);
+
+            Set<IndexInfo> indexInfos = new HashSet<IndexInfo>();
+            if (mapContainer.isGlobalIndexEnabled()) {
+                // global-index
+                for (Index index : mapContainer.getIndexes().getIndexes()) {
+                    indexInfos.add(new IndexInfo(index.getAttributeName(), index.isOrdered()));
+                }
+            } else {
+                // partitioned-index
+                for (PartitionContainer partitionContainer : mapServiceContext.getPartitionContainers()) {
```

Comment:
How come we go over all partition containers and not only the one we're replicating?
```
} else {
                // partitioned-index
                final Indexes indexes = mapContainer.getIndexes(container.getPartitionId());
                if (indexes != null && indexes.hasIndex()) {
                    for (Index index : indexes.getIndexes()) {
                        indexInfos.add(new IndexInfo(index.getAttributeName(), index.isOrdered()));
                    }
                }
            }
```
Is the reason the same as in `com.hazelcast.map.impl.operation.PostJoinMapOperation#addMapIndex`?

Reply:
It was an anti-entropy kind of thing, but seeing your comment I removed it, since it may be really confusing, when you look at it, and there's no significant need for this kind of anti-entropy...